winml/lib/Api/ImageFeatureValue.cpp

```diff
@@ -519,13 +513,14 @@ HRESULT ImageFeatureValue::GetOrtValue(WinML::BindingContext& context, OrtValue*
   OrtAllocator* dml_allocator;
   WINML_THROW_IF_FAILED(m_adapter->GetProviderAllocator(provider, &dml_allocator));
 
+  std::unique_ptr<OrtAllocator>ort_allocator_temp(dml_allocator);
```

Comment:
Why is ort_allocator_temp needed? It's constructing a unique_ptr to OrtAllocator from dml_allocator, which is a pointer to OrtAllocator, and then calling release(). You could use dml_allocator directly, right?

Reply:
And it's dangerous, if you this function returns or throws before you call release() it would try to `delete` the pointer, which is undefined behavior (it must call `m_adapter->FreeProviderAllocator` instead of `delete`)

Reply:
good point! I think I should use the Ort::Allocator I created instead

---
In reply to: [358565189](https://github.com/microsoft/onnxruntime/pull/2653#discussion_r358565189) [](ancestors = 358565189)